torch/csrc/api/include/torch/nn/modules/embedding.h

```diff
@@ -35,14 +69,39 @@ class TORCH_API EmbeddingImpl : public torch::nn::Cloneable<EmbeddingImpl> {
   /// `indices` supplied and returns the result.
   Tensor forward(const Tensor& indices);
 
+  static EmbeddingImpl& from_pretrained(Tensor embeddings, bool freeze = true, c10::optional<int64_t> padding_idx = c10::nullopt,
+          c10::optional<float> max_norm = c10::nullopt, float norm_type = 2., bool scale_grad_by_freq = false, bool sparse = false);
+
   /// The `Options` used to configure this `Embedding` module.
   /// Changes to `EmbeddingOptions` *after construction* have no effect.
   EmbeddingOptions options;
 
-  /// The embedding table.
+  /// The embedding table
   Tensor weight;
 };
 
+class TORCH_API EmbeddingBagImpl() : public torch::nn::Cloneable<EmbeddingBagImpl> {
+  public:
+    EmbeddingBagImpl(int64_t num_embeddings, int64_t embedding_dim)
+      : EmbeddingBagImpl(EmbeddingBagOptions(num_embeddings, embedding_dim)) {}
+    explicit EmbeddingBagImpl(EmbeddingBagOptions options);
+
+    void reset() override;
+
+    /// Pretty prints the `EmbeddingBag` module into the given `stream`.
+    void pretty_print(std::ostream& stream) const override;
+
+    std::tuple<Tensor, Tensor, Tensor, Tensor> forward(const Tensor& input, c10::optional<torch::Tensor> offsets = c10::nullopt,
+      c10::optional<torch::Tensor> per_sample_weights = c10::nullopt);
+
+    static EmbeddingBagImpl& EmbeddingBagImpl::from_pretrained(Tensor embeddings, bool freeze = true, c10::optional<float> mex_norm = c10::nullopt,
+       float norm_type = 2., bool scale_grad_by_freq = false, string mode = "sum", bool sparse = false);
```

Comment:
ditto here: we would need to move `from_pretrained` to `EmbeddingBag` class, so that people can do `torch::nn::EmbeddingBag::from_pretrained(...)`.

Reply:
ditto for using `EmbeddingBagOptions` as well.